src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,89 @@
 package core.basesyntax;
 
+import java.util.LinkedList;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT = 1 << 4;
+    private static final float LOAD_FACTOR = 0.75f;
+    private LinkedList<Entry<K,V>>[] buckets;
+    private V null1;
+    private int size;
+
+    public MyHashMap() {
+        buckets = new LinkedList[DEFAULT];
+        for (int i = 0; i < DEFAULT; i++) {
+            buckets[i] = new LinkedList<>();
+        }
+    }
+
+    private static class Entry<K, V> {
+        private K key;
+        private V value;
+
+        Entry(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            null1 = value;
+            return;
+        }
+        int index = getBucketIndex(key);
+        LinkedList<Entry<K,V>> bucket = buckets[index];
+        for (Entry<K,V> entry : bucket) {
+            if (entry.key.equals(key)) {
+                entry.value = value;
+                return;
+            }
+        }
+        bucket.add(new Entry<>(key, value));
```

Comment:
Use the diamond operator with type parameters when creating a new 'Entry' instance to maintain type safety. The correct instantiation should be 'new Entry<>(key, value)' or 'new Entry<K, V>(key, value)'.